context-lab/src/App.js

```diff
@@ -0,0 +1,36 @@
+import React, {useState, createContext } from "react";
+import NavBar from "./NavBar/NavBar";
+import Clock from "./Components/Clock";
+import ClockProviders from './Providers/ClockProvider'
+import Weather from "./Components/Weather";
+import Weather from './Providers/WeatherProvider'
+import { Route, Switch } from "react-router-dom";
+import Home from "./Components/Home";
+export const ThemeContext = createContext('light')
+function App() {
+  const [theme, setTheme] =useState('light')
+  const toggleTheme = ()=>{
+    setTheme(theme === 'light' ? 'dark': 'light')
+  }
+  return (
+    <div className="App">
+      <ThemeContext.Provider value ={ThemeContext, toggleTheme}>
```

Comment:
You might want to have the provider outside in its own file then just wrap the whole app in it